src/AggregateFunctions/BFloat16Histogram.h

```diff
@@ -0,0 +1,166 @@
+#pragma once
+
+#include <IO/ReadBuffer.h>
+#include <IO/VarInt.h>
+#include <IO/WriteBuffer.h>
+#include <Common/HashTable/HashMap.h>
+#include <Common/PODArray.h>
+#include <common/types.h>
+#include <ext/bit_cast.h>
+
+namespace DB
+{
+
+template <typename Value>
+struct BFloat16Histogram
+{
+    using bfloat16 = UInt16;
+    using Data = HashMap<bfloat16, size_t>;
```

Comment:
I don't understand, why we use *both* hash table and array?

Reply:
I thought hash table is enough.

Reply:
> Linear array of values can be used if the number of values is very small, similar to QuantileTimingTiny.

Reply:
See also quantileExactWeighted.

Reply:
Array used to avoid sorting array every time 'get' called. Is it useless?